src/library/searchquery.cpp

```diff
@@ -166,6 +166,26 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+bool NullTextFilterNode::match(const TrackPointer& pTrack) const {
+    for (const auto& sqlColumn: m_sqlColumns) {
+        QVariant value = getTrackValueForColumn(pTrack, sqlColumn);
+        if (!value.isValid() || !qVariantCanConvert<QString>(value)) {
+            continue;
+        }
+        // only use the major coloumn
+        return value.toString().isEmpty();
+    }
+    return false;
+}
+
+QString NullTextFilterNode::toSql() const {
+    for (const auto& sqlColumn: m_sqlColumns) {
```

Comment:
This loop is confusing if you just want to pick the first column from a non-empty list.